src/AppBundle/Facebook/ImageFilter.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace AppBundle\Facebook;
+
+use Imagine\Image\ImageInterface;
+use Imagine\Image\ImagineInterface;
+use Imagine\Image\Point;
+use Symfony\Component\Finder\Finder;
+
+class ImageFilter
+{
+    private $imagine;
+
+    private $watermarks;
+
+    public function __construct(ImagineInterface $imagine, string $watermarksPath)
+    {
+        $this->imagine = $imagine;
+
+        $this->watermarks = $this->generateWatermarks($watermarksPath);
```

Comment:
I wouldn't do this in the construct method but when needed only. The construct method must never perform any complex logic such as hitting the filesystem. Construct method should only initialize the state of the object.